src/utils.rs

```diff
@@ -218,6 +218,44 @@ pub fn collect_prob_dist<E: Event>(
     Ok(prob_dist)
 }
 
+pub fn filter_by_threshold<E: Event>(
+    calls: &bcf::Reader,
+    threshold: &f64,
+    out: &mut bcf::Writer,
+    events: &[E],
+    vartype: &model::VariantType
+) -> Result<(), Box<Error>> {
+    let mut record = bcf::Record::new();
+    let tags = events.iter().map(|e| e.tag_name("PROB")).collect_vec();
+    let lp_threshold = LogProb::from( PHREDProb( *threshold + 0.000000001) ); // the manual epsilon is required, because the threshold output by `control-fdr` has some digits cut off, which can lead to the threshold being lower than the values reread from the BCF record only due to a higher precision
+    loop {
+        if let Err(e) = calls.read(&mut record) {
+            if e.is_eof() {
+                return Ok(());
+            } else {
+                return Err(Box::new(e));
+            }
+        }
+
+        let variants = (utils::collect_variants(&mut record, false, false, None, false))?;
+        let mut events_prob_sum = LogProb::ln_zero();
+        for tag in &tags {
+            if let Some(event_probs) = (record.info(tag.as_bytes()).float())? {
+                //tag present
+                for (variant, event_prob) in (&variants).into_iter().zip(event_probs.into_iter()) {
+                    if let Some(ref variant) = *variant {
+                        if !variant.is_type(vartype) || event_prob.is_nan() {
+                            continue;
+                        }
+                        events_prob_sum = events_prob_sum.ln_add_exp( LogProb::from( PHREDProb( *event_prob as f64 ) ) );
```

Comment:
Numerically it should be more stable to collect all probs into a vec and then apply `ln_sum_exp`.

Reply:
Done, thanks for the hint. This might also solve a problem I saw on a bigger dataset, where the assertion of `p>=0.0` failed -- I am hoping this was a numerical issue and will double check if this now goes away.